js/interface.js

```diff
@@ -287,6 +273,26 @@ Fliplet.Widget.onSaveRequest(function() {
   save(true);
 });
 
+Fliplet.Widget.onCancelRequest(function() {
+  if (emailTemplateAddProvider) {
+    emailTemplateAddProvider.close();
+    emailTemplateAddProvider = null;
+    emailTemplateAddProvider.forwardCancelRequest();
+    return;
+  }
+  if (providerInstance) {
+    providerInstance.close();
+    providerInstance = null;
+    Fliplet.Studio.emit('widget-save-label-update', {
+      text: 'Save & Close'
+    });
+    Fliplet.Widget.toggleCancelButton(true);
+    Fliplet.Widget.toggleSaveButton(true);
+    Fliplet.Widget.info('');
+    providerInstance.forwardCancelRequest();
```

Comment:
you can't do this because `providerInstance` is set to `null` above. did you test this?

Reply:
@squallstar I did and it worked, and I guess it worked because the file picker provider doesn't use the cancel request, the link takes care of closing it. 

Just gave it another test with console opened and there was an error from that line.
Moved the `forwardCancelRequest()` up.

Tested and working!